Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/EditorVertexSelection.cpp

```diff
@@ -57,13 +61,131 @@ namespace std
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(
+            actionManagerInterface,
+            "EditorVertexSelection - could not get ActionManagerInterface on RegisterActions.");
+
+        auto hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+        AZ_Assert(
+            hotKeyManagerInterface,
+            "EditorVertexSelection - could not get HotKeyManagerInterface on RegisterActions.");
+
+        // Duplicate Vertices
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.vertexSelection.duplicate";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = s_duplicateVerticesTitle;
+            actionProperties.m_description = s_duplicateVerticesDesc;
+            actionProperties.m_category = "Vertex Selection";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                        {
+                            EditorVertexSelectionRequestBus::Event(
+                                entityComponentIdPair, &EditorVertexSelectionRequests::DuplicateSelectedVertices);
+                        }
+                    );
+                }
+            );
+
+            hotKeyManagerInterface->SetActionHotKey(actionIdentifier, "Ctrl+D");
+        }
+
+        // Delete Vertices
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.vertexSelection.delete";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = s_deleteVerticesTitle;
+            actionProperties.m_description = s_deleteVerticesDesc;
+            actionProperties.m_category = "Vertex Selection";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
```

Comment:
So, technically this function should check whether the active component is of the correct type before blasting off the bus call. We skip it here since that bus simply won't be handled for an `EntityComponentIdPair` pointing to a component of the wrong type.

You're correct that we currently don't use it, but I'm expecting it to be useful in many cases. We can remove it and re-introduce it down the line if it actually becomes needed.

Reply:
Okay understood, better to have it and not need it than need it and not have it 😝